data/src/main/java/com/microsoft/azure/kusto/data/exceptions/DataWebException.java

```diff
@@ -6,41 +6,24 @@
 import org.apache.http.HttpResponse;
 import org.json.JSONObject;
 
-public class DataWebException extends Exception{
+public class DataWebException extends WebException {
 
-    private final HttpResponse httpResponse;
-    private OneApiError apiError;
+    private final OneApiError apiError;
 
-    public HttpResponse getHttpResponse() { return httpResponse; }
+    public DataWebException(String message, HttpResponse httpResponse, Throwable cause) {
+        super(message, httpResponse, cause);
+        this.apiError = OneApiError.fromJsonObject(new JSONObject(getMessage()).getJSONObject("error"));
```

Comment:
This can throw - better to keep it in the method then throw from the c'tor

Reply:
Done